### PR TITLE
Fix hardcoded group size in buffer tests

### PR DESCRIPTION
**Pull request by aleksmesh**
Request of work items adjusts to the available capabilities of underlying device in buffer tests.

Function to query available amount of work items moved from tests/group_functions/group_functions_common.h to
tests/common/get_group_range.h.

### CODE REVIEW

File: tests/buffer/buffer_api_common.h [hunk 1]
```diff
@@ -480,24 +481,27 @@ class check_buffer_linearization {
...
-                              sycl::range<1>{size});
-    sycl::nd_range<2> range2d(sycl::range<2>{size, size},
-                              sycl::range<2>{size, size});
-    sycl::nd_range<3> range3d(sycl::range<3>{size, size, size},
-                              sycl::range<3>{size, size, size});
-    // clang-format on
+    auto q = util::get_cts_object::queue();
+    sycl::range<1> range1d = sycl_cts::util::work_group_range<1>(q);
+    sycl::range<2> range2d = sycl_cts::util::work_group_range<2>(q);
+    sycl::range<3> range3d = sycl_cts::util::work_group_range<3>(q);
```
Comment (bader) [inline tests/buffer/buffer_api_common.h:487]:
```suggestion
    sycl::range<1> range1d = sycl_cts::util::work_group_range<1>(q, size);
    sycl::range<2> range2d = sycl_cts::util::work_group_range<2>(q, size * size);
    sycl::range<3> range3d = sycl_cts::util::work_group_range<3>(q, size * size * size);
```
We don't need to use `max_work_item_sizes` to test buffer linearization functionality.

File: tests/common/get_group_range.h [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * @brief Provides range for maximal size work-group
+ *        supported by device of a given queue.
+ *        Multidimentional work-group range is made
+ *        as hyper-cybic as possible
+ * @tparam Dimensions Dimension to use for group instance
+ */
+template <int Dimensions>
+sycl::range<Dimensions> work_group_range(
+    sycl::queue queue,
+    uint64_t work_items_limit = std::numeric_limits<uint64_t>::max()) {
```
Comment (bader) [inline tests/common/get_group_range.h:58]: Shouldn't we use `size_t` here? `max_work_group_size` is `size_t`.

File: tests/buffer/buffer_api_common.h [hunk 3]
```diff
@@ -480,24 +481,27 @@ class check_buffer_linearization {
...
+
+    sycl::nd_range<1> nd_range1d(range1d, range1d);
+    sycl::nd_range<2> nd_range2d(range2d, range2d);
+    sycl::nd_range<3> nd_range3d(range3d, range3d);
```
[the remark this answers is not in the document]
Comment (aleksmesh) [reply tests/buffer/buffer_api_common.h]: Fixed it. Thanks for an advice.

File: tests/common/get_group_range.h [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  return sycl::range<3>{1, 1, 1};
+}
+
+/**
+ * @brief Provides range for maximal size work-group
+ *        supported by device of a given queue.
+ *        Multidimentional work-group range is made
+ *        as hyper-cybic as possible
+ * @tparam Dimensions Dimension to use for group instance
+ */
```
[the remark this answers is not in the document]
Comment (aleksmesh) [reply tests/common/get_group_range.h]: Typo fixed. Thanks.

File: tests/common/get_group_range.h (hunk 2 above)
Comment (aleksmesh) [reply tests/common/get_group_range.h]: You're right. Fixed it.

File: tests/buffer/buffer_api_common.h (hunk 1 above)
Comment (aleksmesh) [reply tests/buffer/buffer_api_common.h]: FIxed it. Thanks!

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**